RecoEgamma/EgammaIsolationAlgos/src/PFBlockBasedIsolation.cc

```diff
@@ -0,0 +1,164 @@
+#include "RecoEgamma/EgammaIsolationAlgos/interface/PFBlockBasedIsolation.h"
+#include <cmath>
+#include "DataFormats/Math/interface/deltaR.h"
+
+
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/Common/interface/RefToPtr.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlock.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementCluster.h"
+
+//--------------------------------------------------------------------------------------------------
+
+PFBlockBasedIsolation::PFBlockBasedIsolation() {
+  // Default Constructor.
+}
+
+
+
+
+//--------------------------------------------------------------------------------------------------
+PFBlockBasedIsolation::~PFBlockBasedIsolation()
+{
+
+}
+
+
+void  PFBlockBasedIsolation::setup ( const edm::ParameterSet& conf ) {
+
+  coneSize_             = conf.getParameter<double>("coneSize");  
+
+}
+
+
+std::vector<reco::PFCandidateRef>  PFBlockBasedIsolation::calculate(math::XYZTLorentzVectorD p4, const reco::PFCandidateRef pfEGCand, const edm::Handle<reco::PFCandidateCollection> pfCandidateHandle) {
+  
+  std::vector<reco::PFCandidateRef> myVec;
+  
+  math::XYZVector candidateMomentum(p4.px(),p4.py(),p4.pz());
+  math::XYZVector candidateDirection=candidateMomentum.Unit();
+
+  const reco::PFCandidate::ElementsInBlocks& theElementsInpfEGcand = (*pfEGCand).elementsInBlocks();
+  reco::PFCandidate::ElementsInBlocks::const_iterator ieg = theElementsInpfEGcand.begin();
+  const reco::PFBlockRef egblock = ieg->first;
+
+
+  unsigned nObj = pfCandidateHandle->size();
+  for(unsigned int lCand=0; lCand < nObj; lCand++) {
+
+    reco::PFCandidateRef pfCandRef(reco::PFCandidateRef(pfCandidateHandle,lCand));
+
+    float dR = 0.0;
+    if( coneSize_ < 10.0 ) {
```

Comment:
This 10.0 looks like a magic number, which CMS coding rules frown upon. Could you replace it with an appropriately named const float?